### PR TITLE
Merging save into cli/dev

### PR DESCRIPTION
**Pull request by dbarer**
Just want someone to do a quick look to make sure the logic here makes sense. Need to have the updated save function from checkpointing to actually test that it works.

### CODE REVIEW

File: src/cli/include/validate.h [hunk 1]
```diff
@@ -77,6 +77,18 @@ bool validate_prep(cmd *c);
  */
 bool validate_ind_objects(cmd *c);
 
+/*
+ * Checks that a specified filetype is a .dat file
+ *
+ * Parameters:
+ * - A string of the filneame
```
Comment (Tianlel) [inline src/cli/include/validate.h]: typo

File: src/cli/include/operations.h [hunk 2]
```diff
@@ -55,6 +55,19 @@ char *help_operation(char *tokens[TOKEN_LIST_SIZE], game_t * game);
...
+/*
+ * Saves the current state of a game to a .dat file
+ * Prints an Error message if filename improperly specified
+ *
+ * Paramters:
+ * - tokens: parsed input string
+ * - pointer to game to be saved,
+ *
+ * Returns:
+ * - A NULL pointer
```
Comment (Tianlel) [inline src/cli/include/operations.h]: should be error message or success message?

File: src/cli/src/operations.c [hunk 3]
```diff
@@ -24,13 +24,15 @@ char *hist_operation(char *tokens[TOKEN_LIST_SIZE], game_t * game)
...
 char *save_operation(char *tokens[TOKEN_LIST_SIZE], game_t *game)
 {
   if(tokens[1] == NULL){
     printf("Save Error, No filename specified");
   }
-  int sv = save(game, tokens[1]);
+  if (validate(tokens[1]) == true){
+    int sv = save(game, tokens[1]);
+  }
   return NULL;
```
[the remark this answers is not in the document]
Comment (dbarer) [reply src/cli/src/operations.c]: Checkpointing handles all the error and success messages inside their save function. I just needed to add the error case where the filename is improperly formatted.

**Comment by Tianlel** [comment]
Closing statement:
To successfully use the save function, we need to build under the main directory. However, right now checkpointing has dummy structures that conflict with the game state structs. So we commented out the newly implemented parts for now. These parts can be used and tested as soon as checkpointing updates their module.

**Comment by lydiafilipe** [comment]
Issue Score: ✓

Comments:
Good work overall! However the title and summary are a bit vague, and you shouldn't be merging without all reviewers' explicit approval

**Pull request merged**